src/WebJobs.Script.WebHost/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -124,5 +126,18 @@ public static class FunctionMetadataExtensions
 
         private static Uri GetFunctionHref(string functionName, string baseUrl) =>
             new Uri($"{baseUrl}/admin/functions/{functionName}");
+
+        private static Uri GetFunctionInvokeUrl(string functionName, string baseUrl, HttpRequest request, IRouter router)
+        {
+            if (router == null)
+            {
+                return null;
+            }
+
+            VirtualPathData path = router.GetVirtualPath(
+                new VirtualPathContext(request.HttpContext, new RouteValueDictionary(), new RouteValueDictionary(), functionName));
```

Comment:
This correctly handles custom routes, right? The function name passed in is used to look up the actual route in the route table?

If a custom route with route constraints is used, this URL will include those?

Reply:
This needs to be properly validated, and we need tests for it.

Reply:
This will not return a path if we're not passing satisfying route conditions (route parameters, constraints, etc.). So as is, this will only return a route if a simple route is being used and it happens to match the context for the API being called. So if a function only supports a POST, this wouldn't work.

As is, this isn't an appropriate API to use for this.